wkcuber/api/Layer.py

```diff
@@ -107,12 +118,12 @@ class WKLayer(Layer):
     mags: Dict[str, WKMagDataset]
 
     def add_mag(
-        self,
-        mag: Union[str, Mag],
-        block_len: int = None,
-        file_len: int = None,
-        block_type: int = None,
+        self, mag: Union[int, str, list, tuple, np.ndarray, Mag], **kwargs: Any
     ) -> MagDataset:
+        block_len: int = kwargs.get("block_len", None)
+        file_len: int = kwargs.get("file_len", None)
+        block_type: int = kwargs.get("block_type", None)
+
```

Comment:
Unfortunately, I had to change the signature of the method to add this method to the base class. This was not an issue in #249 because this method was only called in the tests, which were not covered by `./typecheck`

Reply:
Probably fine for now :+1: That mostly means that the classes are not interchangeable anymore, so this is mostly a question if we should change something about the API design. Not sure though, let's have a quick call about this :calling: